src/Sylius/Component/Promotion/Validator/InstructionValidatorInterface.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Lakion package.
+ *
+ * (c) Lakion
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Promotion\Validator;
+
+use Sylius\Component\Promotion\Generator\InstructionInterface;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+interface InstructionValidatorInterface
+{
+    /**
+     * @param InstructionInterface $instruction
+     *
+     * @return bool
+     */
+    public function isGenerationPossible(InstructionInterface $instruction);
```

Comment:
We should split this into 3 different classes -> _*Validator_, _Checker_ and _Generator_

_Generator_ will provide generation, and number of allowed permutations.
_Checker_ will have custom assertion (50% of possible permutations)
_Validator_ will build constraint violation